### DISCUSSION TITLE
Keyboard shortcut for "Import Icon from Owning Library..." action

### DISCUSSION THREAD

**Discussion by RobustoSystems** [ni/labview-icon-editor#295] [Ideas]
As a LabVIEW user, I would appreciate being able to more quickly execute the "Import Icon from Owning Library..." action.

<img width="746" alt="1 (edited)" src="https://github.com/ni/labview-icon-editor/assets/34945974/c502ede2-9744-4c32-a4fd-a542fc301a30">

This is a repetitive action that I perform several times each day, sometimes dozens of times a day.

I would be happy with any key combination chosen for this action.

Side-note: It would be great if, when moving a VI or CTL from one owner to another, the LabVIEW IDE would be able to automatically update the "banner" of the VI with that of the new owning lvclass or lvlib. This feature would remove 90% of the instances where I need to use the "Import Icon from Owning Library..." action. But I know that this is not the place to ask for this feature.

Thanks!

**Comment by RobustoSystems** [comment]
Optional feature extension:
During its loading/start-up procedure the Icon Editor could detect if the VI/CTL has the "wrong" banner (banner does not match current owner banner). If that is the case, the Icon Editor could display a two button dialogue containing a message such as:

"This VI does not use the banner of its owner, "name of owner here". Would you like the banner to be applied?"

If the user selects "Yes" (or hits Enter), the Icon Editor would execute the "Import Icon from Owning Library..." action for the user.

Benefits:
- This functionality would act like a mini VI analyser. It would help catch this issue.
- Clicking Yes (or hitting Enter) would be even quicker than executing the keyboard shortcut.
Downsides:
- Performing the check may slow down the start-up sequence. This may be a significant drawback.

**Comment by crossrulz** [comment]
If going the dialog on load, also enable a checkbox to "Always Apply".

**Comment by svelderrainruiz** [comment]
For the purpose of closing this PR, i will develop the keyboard shortcut on the icon editor and will open a new issue that explores the route of automatically applying the banner from the parent library as described on this issue's comments

**Comment by RobustoSystems** [comment]
Thanks for looking into this. It's much appreciated.

Kind regards,
Petru Tarabuta, Robusto Systems Ltd, UK

**Comment by RobustoSystems** [comment]
Hi Sergio,

I notice that you marked this issue as completed 3 days ago. Just to double-check, is this functionality actually implemented now?

If so, what keyboard action was assigned? And who implemented the feature (for my own curiosity)?

Thanks,
Petru

FAO: @svelderrainruiz

**Comment by svelderrainruiz** [comment]
@RobustoSystems  My mistake. 

I designed the functionality, it is currently on the branch from this github issue, but i would rather have someone from the community take this issue.

What i designed can be used by a feature developer if he/she wants to consider what i did on that branch.

Once this gets approved by the steering committee

**Comment by crossrulz** [comment]
There are two things in here:
1) A shortcut to import from the owning library
2) An automatic import of some kind.

I moved this to a discussion so we can break these two ideas into separate issues once implementation is figured out.

**Comment by crossrulz** [comment]
Since adding a shortcut is super easy (ie edit the RTM file), let's figure that one out. The only real question is what to make the shortcut. My current thought is Ctrl+I (I for Import) or Ctrl+L (L for Library, note that Ctrl+Shift+L is already used to lock the context help). I could easily be talked into a different shortcut as long as it makes reasonable sense.

**Reply by RobustoSystems** [reply]
FWIW, I would be equally happy with either Ctrl+I or Ctrl+L.
